### ISSUE TITLE
Update functions best practice

### ISSUE THREAD

**Issue by GantMan** [infinitered/ignite#218]
Standard now allows 
```javascript
  constructor (props) {
    super(props)
    this.handleChangeEmail = this.handleChangeEmail.bind(this)
  }

  handleChangeEmail (text) {
    this.setState({ email: text })
  }
```

to become
```javascript
  handleChangeEmail = (text) => {
    this.setState({email: text})
  }
```

which should properly bind `this`.   We should update accordingly in examples.

**Comment by GantMan** [comment]
@siakaramalegos - this would be an excellent PR

**Comment by siakaramalegos** [comment]
on it! :+1:

**Comment by ahmed1490** [comment]
how about double colon?
https://gist.github.com/islahul/73f99302a721a714c002#file-react-double-colon-meaning-es6-js-L34

**Comment by GantMan** [comment]
@ahmed1490 - great find.  This is the first time I've seen the double colon.

Does it have any adverse effects?   What happens if we try to bind some other parameter to something that's been double coloned?

@skellock  - I'd love to drag you into this.   The task is getting slightly philosophical now.

My gut reaction, double colon is cool, but it's just "neat".  It begs more questions than it answers.
The function syntax we're proposing, looks clean.

thoughts?

**Comment by ahmed1490** [comment]
@GantMan I have been using it for simple one - one binds. 
- Does it have any adverse effects?
  - Works fine until now for me!

- What happens if we try to bind some other parameter to something that's been double coloned?
  - Not sure I get your question.

**Comment by skellock** [comment]
I'm not a fan of `::`.  

`.bind` is notorious for confusing people.  I'm not convinced that sugaring it to `::` will help with that.

I respect that `::` is an attempt to de-funk `.bind`, but I think a better way out would be to remove the need entirely.

Here's more info on this:

http://babeljs.io/docs/plugins/transform-function-bind/

**Comment by GantMan** [comment]
closed via #229
